redis/src/main/scala/zio/redis/Output.scala

```diff
@@ -199,10 +200,267 @@ object Output {
         throw ProtocolError(s"$text isn't unit.")
   }
 
+  case object GeoOutput extends Output[Chunk[LongLat]] {
+    override protected def tryDecode(text: String) =
+      if (text.startsWith("*"))
+        parse(text)
+      else
+        throw ProtocolError(s"$text isn't scan output.")
+
+    private[this] def parse(text: String): Chunk[LongLat] =
+      unsafeReadCoordinates(text)
+  }
+
+  case object GeoRadiusOutput extends Output[Chunk[GeoView]] {
+    override protected def tryDecode(text: String) =
+      if (text.startsWith("*"))
+        parse(text)
+      else
+        throw ProtocolError(s"$text isn't scan output.")
+
+    private[this] def parse(text: String): Chunk[GeoView] = {
+      val len    = unsafeReadLength(text)
+      var output = Array.ofDim[GeoView](len)
+
+      if (containsInnerArray(text)) {
+        val pos          = skipToNext(text, 0)
+        val coordinates  = unsafeReadCoordinates(text)
+        val containsHash = containsGeoHash(text)
+
+        if (coordinates.isEmpty && !containsHash)
+          output = geoViewWithDistance(text, pos, len)
+        else if (containsHash)
+          output = geoViewWithHash(text, pos, len, coordinates)
+        else
+          output = geoViewWithoutHash(text, pos, len, coordinates)
+
+        Chunk.fromArray(output)
+      } else unsafeReadMembers(text, len)
+    }
+
+    private[this] def geoViewWithDistance(text: String, start: Int, len: Int): Array[GeoView] = {
+      var idx    = 0
+      var pos    = start
+      val output = Array.ofDim[GeoView](len)
+
+      while (idx < len) {
+        val member = unsafeReadString(text, pos)
+
+        pos = skipToNext(text, pos)
+
+        output(idx) = GeoView(member, Some(unsafeReadString(text, pos).toDouble), None, None)
+        idx += 1
+
+        if (idx < len) pos = skipToNext(text, pos)
+      }
+
+      output
+    }
+
+    private[this] def geoViewWithHash(
+      text: String,
+      start: Int,
+      len: Int,
+      coordinates: Chunk[LongLat]
+    ): Array[GeoView] = {
+      var idx    = 0
+      var pos    = start
+      val output = Array.ofDim[GeoView](len)
+
+      while (idx < len) {
+        val member = unsafeReadString(text, pos)
+
+        while (text.charAt(pos) != '\n') pos += 1
+        pos += 1
+
+        val containsDistance = text.charAt(pos) != ':'
+        val longLat          = if (coordinates.isEmpty) None else Some(coordinates(idx))
+
+        if (!containsDistance) {
```

Comment:
You're using double negation here, swap it :). Also, if is an expression, see remark above.